packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -775,7 +773,7 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		this.disposed = true;
 		this.#transaction.dispose();
 		this.purgeRevertibles();
-		this._branch.dispose();
+		this.#transaction.activeBranch.dispose();
```

Comment:
It's weird to call this after `this.#transaction.dispose();`. Also, don't we need to ensure that the base branch is also disposed even when this disposal occurs during a transaction. It's not clear to me whether that's the case right now.

Reply:
Good catch, I moved the dispose of the branch to come first and changed it to always dispose the base branch, not the active branch.

The transaction branch, if there is one, will get disposed as part of disposing the transaction.